web/src/connectors/documents-page-control-connector/use-notify-processing.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react';
+import { NotificationTypes, showNotification } from 'shared/components/notifications';
+
+const notifyProcessing = () => {
+    showNotification({
+        type: NotificationTypes.hint,
+        text: 'Your file is still processing. This may take a while'
+    });
+};
+
+export const useNotifyProcessing = (isUploaded: boolean) => {
+    useEffect(() => {
+        let delay = 10000;
+        let timeoutId: number | null = null;
+        let firstNotificationShown = false;
+
+        const showNextNotification = () => {
+            if (isUploaded) {
+                if (!firstNotificationShown) {
+                    timeoutId = setTimeout(() => {
+                        firstNotificationShown = true;
+                        showNextNotification();
```

Comment:
Am I right that after 10 secs we don't call "notifyProcessing"?

Reply:
No, on the first round, `firstNotificationShown` is false, so the first timeout is set calling `showNextNotification` on the timeout callback. and setting `firstNotificationShown` as true

after 10 secs, `showNextNotification` is called again because of the last timeout, but this time `firstNotificationShown` is true (it was set as true 10 secs before this round), and we fall into the else block which immediately calls the `notifyProcessing` function and sets a new timeout with double delay

Reply:
> No, on the first round, firstNotificationShown is false, so the first timeout is set calling showNextNotification on the timeout callback. and setting firstNotificationShown as true

But why on the first round we don't show notification?